### PR TITLE
Checkpoint 3 Review

### PR DESCRIPTION
**Pull request by megwali**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  validates_presence_of :firstname, :lastname, :email
+  validates_uniqueness_of :email
+  validates_presence_of :password, on: :create
+
+  before_save :generate_iss_number
+
+  private
+
+  def generate_iss_number
```
Comment (kayodeadeniyi) [inline app/models/user.rb]: @andela-megwali What's this number used for?

File: config/application.rb [hunk 2]
```diff
@@ -22,5 +22,6 @@ class Application < Rails::Application
 
     # Do not swallow errors in after_commit/after_rollback callbacks.
     config.active_record.raise_in_transactional_callbacks = true
+    config.autoload_paths << Rails.root.join("lib")
```
Comment (kayodeadeniyi) [inline config/application.rb:25]: @andela-megwali This ought to be autoloaded by default.

File: db/migrate/20161028104813_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateUsers < ActiveRecord::Migration
+  def change
+    create_table :users do |t|
+      t.string :firstname
+      t.string :lastname
+      t.string :email
+      t.string :password_digest
+      t.string :iss
```
Comment (kayodeadeniyi) [inline db/migrate/20161028104813_create_users.rb]: @andela-megwali This property has never been descriptive for once. I keep asking what it means

**Comment by kayodeadeniyi** [comment]
@andela-megwali Looks good overall. I had some few comments.

File: app/models/user.rb (hunk 1 above)
Comment (megwali) [reply app/models/user.rb]: JWT has the iss (issuer) claim similar to the exp (expiry) claim which can be used to verify the token's validity.
See: https://tools.ietf.org/html/rfc7519#section-4.1.1
In this app it is being used to invalidate all active tokens when reset using the logout action.
Thanks.

File: db/migrate/20161028104813_create_users.rb (hunk 3 above)
Comment (megwali) [reply db/migrate/20161028104813_create_users.rb]: Explained already.
See: https://tools.ietf.org/html/rfc7519#section-4.1.1
Thanks.

File: config/application.rb (hunk 2 above)
Comment (megwali) [reply config/application.rb:25]: It doesn't seem so in this app. Probably due to the minimalistic use of middleware by rails-api.
  ¯_(ツ)_/¯

File: app/controllers/api/v1/bucketlists_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        @bucketlists = @current_user.bucketlists
+      end
+
+      def set_bucketlist
+        @bucketlist = @bucketlists.find_by(id: params[:id])
+        @bucketlist ||= { error: not_permitted_message }
+      end
+
+      def bucketlist_params
+        params.require(:bucketlist).permit(:name)
```
Comment (megwali) [inline app/controllers/api/v1/bucketlists_controller.rb]: Facilitator Feedback: Require not necessary to improve user experience.

File: app/models/concerns/paginate.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module Paginate
+  def paginate(limit_query, page_query)
+    limit(paginate_limit(limit_query)).order("id ASC").
```
Comment (megwali) [inline app/models/concerns/paginate.rb:3]: Pagination bug on production environment fixed by explicitly ordering the query.

File: app/models/user.rb (hunk 1 above)
Comment (megwali) [reply app/models/user.rb]: Facilitator Feedback: Name should be changed to conform with local style guide and not JWT's.

File: app/serializers/user_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class UserSerializer < ActiveModel::Serializer
+  include SerializerHelper
+
+  attributes :id, :firstname, :lastname, :email, :date_modified, :user_message
```
Comment (megwali) [inline app/serializers/user_serializer.rb]: Facilitator Feedback: Change user_message to message

**Pull request merged**